### DISCUSSION TITLE
Splink 4.0.0: Cannot save completeness chart to PNG

### DISCUSSION THREAD

**Discussion by aalexandersson** [moj-analytical-services/splink#2335] [Q&A] [answered]
I can no longer save off-line charts to PNG for PDF; I had no problem doing this using Splink 3.9.15. I am using Splink 4.0.0 with the default DuckDB backend on Python 3.11.9 for Windows, 

I assume it is a user error (my fault!). It is a new installation of Python and `venv`. It's a new setup because I encountered dependency hell; for details see [discussion issue 2316](https://github.com/moj-analytical-services/splink/discussions/2316). 

The code below illustrates the issue for the completeness chart. I also cannot save a blocking chart to PNG. I do not have a reprex yet. In the requirements.txt file I have, for example, `altair==5.4.0`, `splink=4.0.0,` and `vl-convert-python==1.6.0`.

```
from splink.exploratory import completeness_chart  # 
c_completeness = completeness_chart([df1, df2],
    cols=['simulant_id', 
    'first_name', 'last_name', 'ssn', 'sex', 'address', 'city', 
    'state', 'zip', 'street_number', 'street_name', 'dob'],
     db_api=db_api)  
c_completeness.save("c_completeness.png")                   # 
```

The error message is:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "V:\Testing\Splink_2025\.venv\Lib\site-packages\altair\vegalite\v5\api.py", line 2086, in save
    save(**kwds)
  File "V:\Testing\Splink_2025\.venv\Lib\site-packages\altair\utils\save.py", line 224, in save
    perform_save()
  File "V:\Testing\Splink_2025\.venv\Lib\site-packages\altair\utils\save.py", line 175, in perform_save
    mb_png = spec_to_mimebundle(
             ^^^^^^^^^^^^^^^^^^^
  File "V:\Testing\Splink_2025\.venv\Lib\site-packages\altair\utils\mimebundle.py", line 134, in spec_to_mimebundle
    return _spec_to_mimebundle_with_engine(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "V:\Testing\Splink_2025\.venv\Lib\site-packages\altair\utils\mimebundle.py", line 244, in _spec_to_mimebundle_with_engine
    png = vlc.vegalite_to_png(
          ^^^^^^^^^^^^^^^^^^^^
ValueError: Vega-Lite to PNG conversion failed:
Failed to send SVG conversion request: send failed because receiver is gone
```

**Comment by RobinL** [comment]
Does this work?

```
import altair as alt
df = pd.DataFrame({"x": [2, 3, 4], "y": [5, 5, 3]})
chart = alt.Chart(df).mark_point().encode(x="x", y="y")
chart.save("del.png")
```

If that fails, the issue is with `altair`/`vl-convert-python` as opposed to Splink

**Reply by aalexandersson** [reply, marked as answer]
The issue was with my RStudio IDE, which I use for automated reporting in PDF. Upgrading RStudio IDE from 2024.04.1+748 (outdated) to 2024.04.2+764 (current) solved the issue.

Robin's code also works; just need to first add `import pandas as pd`.
